source/scripting_v3/GTA/World.cs

```diff
@@ -2116,6 +2150,26 @@ public static RelationshipGroup AddRelationshipGroup(string name)
 			return new RelationshipGroup(resultArg);
 		}
 
+		/// <summary>
+		/// Creates a shocking event of specified type at the given position for the given duration.
+		/// </summary>
+		/// <param name="eventType">The type of shocking event peds nearby will perceive.</param>
+		/// <param name="position">The epicenter position of the shocking event.</param>
+		/// <param name="duration">The duration of the shocking event.</param>
+		public static void CreateShockingEvent(EventType eventType, Vector3 position, float duration)
```

Comment:
We must handle this kind of shocking events with a dedicated class because `ADD_SHOCKING_EVENT_AT_POSITION` returns a handle. Return null instead of a new instance if the native returns zero.
Make sure the new class implements `Equatable<T>` (as usual) so collections like `Dictionary` can find elements faster. Unfortunately we can't implement `PoolObject` or even `IDeletable` without direct memory access (well they have too much constraints IMO) since R* didn't bother to expose a native to check if a shocking event exists, but we could implement `INativeValue`.

Also, the third one should be `overrideLifeTime`, where 0 specifies the infinite time and -1 (or any negative one I think) specifies the default life time if the docs says correctly.

The doc of the native:
```
//INFO: 
//PARAM NOTES: eventType must be one of the EVENT_SHOCKING_... events. overrideLifeTime is in seconds and can be 0 for infinite, or left at -1 to use the default for the event type.
//PURPOSE: Add a specific shocking event at a position, and return an event ID if successful, or 0 if nothing was added.
NATIVE FUNC INT ADD_SHOCKING_EVENT_AT_POSITION(EVENT_NAMES eventType, VECTOR pos, FLOAT overrideLifeTime = -1.0) = "0xaa8ea67d77d37c61"
```

Reply:
Understood. I've removed this from this PR and will place it on my to-do list for creating dedicated classes.